iocore/net/quic/QUICFlowController.cc

```diff
@@ -130,11 +151,40 @@ QUICLocalFlowController::forward_limit(QUICOffset offset)
   QUICFlowController::forward_limit(offset);
 
   // Send MAX_(STREAM_)DATA frame
-  if (this->_limit - this->_offset <= this->_threshold) {
+  if (this->_need_to_gen_frame()) {
     this->_frame = this->_create_frame();
   }
 }
 
+int
+QUICLocalFlowController::update(QUICOffset offset)
```

Comment:
This doesn't look correct. If we do this, the credit will be increased immediately without any limit when it receives new data. We should call `forward_limit()` after processing received data. Otherwise flow control doesn't work.